File: src/com/serviceImplement/MaterialCal.java
package com.serviceImplement;

import com.services.MaterialManagement;

import com.Exception.LessThanZeroException;

public class MaterialCal implements MaterialManagement {
	
	DatabaseImplements db=new DatabaseImplements();
	public int cementCal(int count) throws LessThanZeroException   //cement amout calculation count must be positive value
{
		if(count>=0) {
				count=count*db.getRates("cement");
		return count;
	}else{
		throw new LessThanZeroException("Cement count must be positive value");
	}}
	public int sandCal(int count) throws LessThanZeroException //sand amount calculation count must be positive value
{
		if(count>=0) {
		count = count*db.getRates("sand");
		return count;
	}else {
		throw new LessThanZeroException("Sand count must be positive value");
	}}
	public int brickCal(int count) throws LessThanZeroException   //brick amount caliculation count must be positive value
{
		if(count>=0) {
		count=count*db.getRates("brick");
		return count;}else {
			throw new LessThanZeroException("Brick count must be positive value");
		}
	}

}

File: src/com/serviceImplement/MaterialCal.go
package serviceImplement

import "fmt"

type MaterialCal struct {
}

func CreateMaterialCal() MaterialCal {
	return MaterialCal{}
}

// DatabaseImplements db=new DatabaseImplements();
func (MaterialCal) CementCal(count int) int {
	if count >= 0 {
		count := count * GetRates("cement")
		return count
	} else {
		fmt.Println("Cement count must be positive value")
	}
	return 0
}
func (MaterialCal) SandCal(count int) int {
	if count >= 0 {
		count := count * GetRates("sand")
		return count
	} else {
		fmt.Println("Sand count must be positive value")
	}
	return 0
}
func (MaterialCal) BrickCal(count int) int {
	if count >= 0 {
		count := count * GetRates("brick")
		return count
	} else {
		fmt.Println("Brick count must be positive value")
	}
	return 0
}
